src/Host/Client/Impl/Host/RHostBrokerConnector.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.R.Interpreters;
+
+namespace Microsoft.R.Host.Client.Host {
+    public sealed class RHostBrokerConnector : IRHostBrokerConnector {
```

Comment:
So then it should be [internal?] part of the workflow rather than passed to every session operation?